mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapperTest.java

```diff
@@ -0,0 +1,189 @@
+        cassandraAnnotationMapper.updateAnnotation(new SimpleMailboxAnnotationCommand(key, value));
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(new SimpleMailboxAnnotation(buildOneItem()));
+    }
+
+    @Test
+    public void addAnnotationWhenStoredShouldReturnUpdatedAnnotation() throws Exception {
+        Session currentSession = cassandra.getConf();
+        currentSession.execute(
+                insertInto(CassandraAnnotationTable.TABLE_NAME)
+                .value(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid())
+                .value(CassandraAnnotationTable.KEY, "/private/comment")
+                .value(CassandraAnnotationTable.VALUE, "My stored private comment")
+                );
+        
+        SimpleMailboxAnnotationEntryKey key = new SimpleMailboxAnnotationEntryKey("/private/comment");
+        SimpleMailboxAnnotationEntryValue value = new SimpleMailboxAnnotationEntryValue("My private comment", AnnotationValue.string);
+        cassandraAnnotationMapper.updateAnnotation(new SimpleMailboxAnnotationCommand(key, value));
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(new SimpleMailboxAnnotation(buildOneItem()));
+    }
+
+    @Test
+    public void addNilAnnotationWhenNoneStoredShouldReturnEmpty() throws Exception {
+        SimpleMailboxAnnotationEntryKey key = new SimpleMailboxAnnotationEntryKey("/private/comment");
+        cassandraAnnotationMapper.updateAnnotation(new SimpleMailboxAnnotationCommand(key, SimpleMailboxAnnotationEntryValue.NIL));
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(SimpleMailboxAnnotation.EMPTY);
+    }
+
+    @Test
+    public void addNilAnnotationWhenStoredShouldReturnEmpty() throws Exception {
+        Session currentSession = cassandra.getConf();
+        currentSession.execute(
+                insertInto(CassandraAnnotationTable.TABLE_NAME)
+                .value(CassandraAnnotationTable.ID, mailbox.getMailboxId().asUuid())
+                .value(CassandraAnnotationTable.KEY, "/private/comment")
+                .value(CassandraAnnotationTable.VALUE, "My stored private comment")
+                );
+
+        SimpleMailboxAnnotationEntryKey key = new SimpleMailboxAnnotationEntryKey("/private/comment");
+        cassandraAnnotationMapper.updateAnnotation(new SimpleMailboxAnnotationCommand(key, SimpleMailboxAnnotationEntryValue.NIL));
+        assertThat(cassandraAnnotationMapper.getAnnotation()).isEqualTo(SimpleMailboxAnnotation.EMPTY);
+    }
+
+    private Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> buildOneItem() {
+        Map<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue> resultEntries = 
+                new HashMap<MailboxAnnotationEntryKey, MailboxAnnotationEntryValue>();
+        resultEntries.put(new SimpleMailboxAnnotationEntryKey("/private/comment"), 
+                new SimpleMailboxAnnotationEntryValue("My private comment", AnnotationValue.string));
```

Comment:
the SimpleMailboxAnnotation API is definitively too complex : you are building factory method for building a very simple object, that should strike you how you define such a complex API.